packages/catalog-search/src/SearchBox.jsx

```diff
@@ -135,7 +171,7 @@ export const SearchBoxBase = ({
         <SearchSuggestions
           enterpriseSlug={enterpriseSlug}
           autoCompleteHits={autocompleteHits}
-          handleViewAllClick={() => handleSubmit(searchQuery)}
+          handleSubmit={hit => handleSuggestionSubmit(hit.title, hit)}
```

Comment:
when this `handleSubmit` will be called by the View all results button click then there wont be any hit passed? the desired actions would be performed in that case? is this expected and taken care of? what learner portal expects is that when view all is clicked then **whatever the user have typed** will be the new query and all search results below will be updated.

when clicked on `SearchSuggestionItem` according to the logic if we provide override then that would be called along with setting the search query in search box to the title of the item we just clicked, which should happen? because the title isnt the query and we may not want to set the text in search box to the title of item we just hit? is that it or am i missing something?

Reply:
clicking view all will set the original search query, clicking the individual row will set the query as the title of the clicked row.